src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -540,7 +540,7 @@ def model_sharded_context(self) -> Generator[None, None, None]:
         if self.zero_stage_3:
             assert self._config_initialized
 
-            if self.precision_plugin.precision == PrecisionType.HALF:
+            if self.precision_plugin.precision in (PrecisionType.HALF, PrecisionType.MIXED):
```

Comment:
"Mixed" is not meant to be true half precision, so I don't think this is correct: https://github.com/Lightning-AI/lightning/issues/9956